typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -203,4 +219,15 @@ export class EvmERC20WarpRouteReader {
 
     return { name, symbol, decimals, totalSupply: totalSupply.toString() };
   }
+
+  /**
+   * Conditionally sets the log level for a smart provider.
+   *
+   * @param level - The log level to set, e.g. 'debug', 'info', 'warn', 'error'.
+   */
+  protected setSmartProviderLogLevel(level: string) {
+    if (this.provider instanceof HyperlaneSmartProvider) {
```

Comment:
Recommend against instanceof in production code.
Better would be to check for the existence of a `setLogLevel` fn on the provider, even if it requires a ts-ignore here.